Mathlib/NumberTheory/ArithmeticFunction.lean

```diff
@@ -270,7 +286,8 @@ theorem natCoe_mul [Semiring R] {f g : ArithmeticFunction ℕ} :
   simp
 #align nat.arithmetic_function.nat_coe_mul Nat.ArithmeticFunction.natCoe_mul
 
-@[simp, norm_cast]
+@[simp] -- porting note: the tag `norm_cast`, present in mathlib3, causes the linter to complain:
+        -- `badly shaped lemma, lhs must contain at least one coe`
 theorem intCoe_mul [Ring R] {f g : ArithmeticFunction ℤ} :
     (↑(f * g) : ArithmeticFunction R) = f * g := by
```

Comment:
Here, the problem is that Lean 4 interprets RHS as `↑(f * g)` instead of `↑f * ↑g`. You should add an explicit `↑` somewhere in the RHS.

Reply:
I tried this, but apparently no choice of subset of `f, g` on which to put the coercion makes the `norm_cast` linter happy.  Note that it complains about the LHS, not the RHS.

Reply:
It's because the coercion didn't have the `@[coe]` attribute